cfgov/v1/documents.py

```diff
@@ -0,0 +1,201 @@
+from django_elasticsearch_dsl import Document, fields
+from django_elasticsearch_dsl.registries import registry
+
+from v1.models.base import (
+    CFGOVAuthoredPages, CFGOVPageCategory, CFGOVTaggedPages
+)
+from v1.models.learn_page import AbstractFilterPage
+
+
+@registry.register_document
+class FilterablePagesDocument(Document):
+
+    tags = fields.ObjectField(properties={
+        'slug': fields.KeywordField()
+    })
+    categories = fields.ObjectField(properties={
+        'name': fields.KeywordField()
+    })
+    authors = fields.ObjectField(properties={
+        'name': fields.TextField(),
+        'slug': fields.KeywordField()
+    })
+    title = fields.TextField(attr='title')
+    is_archived = fields.KeywordField(attr='is_archived')
+    content = fields.TextField()
+    date_published = fields.DateField(attr='date_published')
+    url = fields.KeywordField()
+    start_dt = fields.DateField()
+    end_dt = fields.DateField()
+    statuses = fields.KeywordField()
+    initial_filing_date = fields.DateField()
+
+    def get_queryset(self):
+        return AbstractFilterPage.objects.live().public()
+
+    def prepare_content(self, instance):
+        try:
+            content_field = instance._meta.get_field('content')
+            value = content_field.value_from_object(instance)
+            content = content_field.get_searchable_content(value)
+            content = content.pop()
+            return content
+        except Exception:
+            return None
+
+    def prepare_url(self, instance):
+        return instance.url
+
+    def prepare_start_dt(self, instance):
+        try:
+            return instance.specific.start_dt
+        except Exception:
+            return None
```

Comment:
Instead of repeating this pattern, could you instead do

```py
return getattr(instance.specific, 'start_dt', None)
```

Using an `except Exception` here means that any errors will be caught, including those that are unrelated to whether the instance has a particular attribute. For example, if there's some database connectivity issue, that would be caught here, where we'd prefer that it get raised.

Reply:
Thats way better! I also dislike repeating try blocks and prefer to implement a more functional accessor. Thanks!